In [1]:
import numpy as np
import pandas as pd

In [3]:
data = pd.read_csv("C:/Users/Dell/Desktop/research dataset/dataset 1/normalized_Data.csv")

In [5]:
data.shape

(58693, 24)

In [7]:
data.columns

Index(['ID', 'Day', 'Incidence', 'Brand', 'Quantity', 'Last_Inc_Brand',
       'Last_Inc_Quantity', 'Price_1', 'Price_2', 'Price_3', 'Price_4',
       'Price_5', 'Promotion_1', 'Promotion_2', 'Promotion_3', 'Promotion_4',
       'Promotion_5', 'Sex', 'Marital status', 'Age', 'Education', 'Income',
       'Occupation', 'Settlement size'],
      dtype='object')

In [8]:
data.head(5)

ID       Day  Incidence  Brand  Quantity  Last_Inc_Brand  \
0  0.0  0.000000        0.0    0.0       0.0             0.0   
1  0.0  0.013717        0.0    0.0       0.0             0.0   
2  0.0  0.015089        0.0    0.0       0.0             0.0   
3  0.0  0.020576        0.0    0.0       0.0             0.0   
4  0.0  0.023320        0.0    0.0       0.0             0.0   

   Last_Inc_Quantity   Price_1   Price_2   Price_3  ...  Promotion_3  \
0                0.0  1.000000  0.953125  0.518519  ...          0.0   
1                0.0  0.836735  0.984375  0.444444  ...          0.0   
2                0.0  0.836735  0.984375  0.444444  ...          0.0   
3                0.0  0.857143  0.984375  0.407407  ...          0.0   
4                0.0  0.857143  0.984375  0.444444  ...          0.0   

   Promotion_4  Promotion_5  Sex  Marital status       Age  Education  \
0          0.0          0.0  0.0             0.0  0.508772   0.333333   
1          0.0          0.0  0.0             0.0  0.508772   0.333333   
2          0.0          0.0  0.0             0.0  0.508772   0.333333   
3          0.0          0.0  0.0             0.0  0.508772   0.333333   
4          0.0          0.0  0.0             0.0  0.508772   0.333333   

     Income  Occupation  Settlement size  
0  0.267851         0.5              0.0  
1  0.267851         0.5              0.0  
2  0.267851         0.5              0.0  
3  0.267851         0.5              0.0  
4  0.267851         0.5              0.0  

[5 rows x 24 columns]

In [9]:
#Using linear regression to predict values
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split

In [13]:
train = data.drop(['Quantity','ID','Day'],axis=1)
test = data['Quantity']
X_train,X_test, y_train, y_test = train_test_split(train, test, test_size= 0.3, random_state= 2)

In [14]:
regr = LinearRegression()

In [15]:
regr.fit(X_train,y_train)

LinearRegression()

In [16]:
pred = regr.predict(X_test)

In [17]:
pred

array([-5.99187532e-04,  1.97785894e-01, -4.61443255e-05, ...,
       -2.59242914e-03, -3.79205770e-03,  4.28975831e-03])

In [18]:
regr.score(X_test,y_test)

0.6563353128893901

Rgression score of 0.6563353128893901 is a lower score for a regression model. to work. Therefore there can be other Supervised Learning Models that may be more fit in this.